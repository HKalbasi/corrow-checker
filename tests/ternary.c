#include <stddef.h>

void *owned malloc(unsigned long size);
void free(void *owned ptr);

int main()
{
    void *p1 = malloc(5);
    1 == 2 ? free(p1) : free(0);
    return 0;
}