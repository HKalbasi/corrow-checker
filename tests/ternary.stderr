Error: Potential leak detected
    ╭─[preproccessed_file.c:1:1]
    │
 31 │     void *p1 = malloc(5);
    │              ─────┬┬────  
    │                   ╰─────── Ownership of the value transferred to this place here
    │                    │      
    │                    ╰────── Ownership of the value transferred to this function here
    │ 
 33 │     return 0;
    │     ────┬────  
    │         ╰────── Function exited without proper destruction of that value
────╯